src/zip.js

```diff
@@ -48,26 +52,53 @@ class Zip {
 
     while (pipeline.length) {
       const current = pipeline.pop()
-      if (this._isFolder(current.localPath)) {
-        // Add folder.
-        entries.push({ localPath: current.localPath, zipPath: current.zipPath + '/', buffer: Buffer.alloc(0) })
-
-        // Walk down the tree.
-        for (const entityName of fs.readdirSync(current.localPath)) {
-          pipeline.push({
-            localPath: current.localPath + '/' + entityName,
-            zipPath: current.zipPath + '/' + entityName
-          })
+
+      while (true) {
```

Comment:
I don't think this is the right solution. The pipeline itself is the main waiting mechanism. Adding another one here that kind-of does the same thing complicates the implementation. I see two possible alternatives:
- Skip the processing of files and folders that were deleted in the meanwhile  (they won't go into a ZIP, so the update will be partial).
- Skip the entire batch from deploying and wait for the next batch - until no errors are present (assuming the bulk change is done).

Reply:
1. it is not possible, because file is deleted right after it started being processed to be added in ZIP.
2. it is not working, no files are processing after this batch. And no updates are uploaded.

Reply:
this while true is just for retrying file, otherwise if it is skipped, it will be never processed for this update, to trigger next update user have to do another update of files or restart aemsync and do another update of files, this is really annoying and took a lot of time.

Believe me first approach i did was to skip files or batch :D and then we had more problems with no files processed at all( skipped files, that caused errors ).

Reply:
this approach already used by team of ~30 developers, and shows only good results.

Reply:
Can you please either add a test case to trigger the issue or provide me with some steps to reproduce? If the file is already in the zip, that should not be a problem. It is only a problem if it is deleted after it's been reported and before it's added to the zip, so skipping it should work. The deletion will be reported in the next pipeline cycle and will end up in the next package.

Whilst it may be fixing the issue for the team, it breaks the separation of concerns and makes the code unnecessarily complex. I would rather have a more graceful solution in place, i.e. waiting for all consecutive changes to finish (by e.g. doing deployments only after a certain amount of time passes from the last reported change.

Reply:
Got it, i will prepare use cases, but all the errors only happens on windows based machines, on linux there is almost no issues at all.

Do you have windows based host for testing?

Reply:
I do, thanks. I am working on version 5, which has a different approach to file processing. I should have something workable next week and it would be great if you could test it on your project.

Reply:
looking forward to test version 5 :)

Reply:
Try this branch:
https://github.com/gavoja/aemsync/tree/feature/v5

There are some backward incompatible changes, see here:
https://github.com/gavoja/aemsync/blob/feature/v5/CHANGELOG.md

Reply:
struggling a lot trying to run v5
![image](https://user-images.githubusercontent.com/19553159/126765564-931386e0-de43-45e1-9df6-edce9a723681.png)
getting this error with different versions of nodejs on windows

Reply:
with version 4 I doesn't getting such error

Reply:
there is warning that 12 version is not supported by watcher, i tried latest 14 LTS version, same errror

Reply:
I don't think it will work like this. Try simply cloning the repository and running `node index.js` from the main folder (that's what `aemsync` binary shortcut runs).

Reply:
I've published it as a pre-release. Do `npm install -g aemsync@next`.

Reply:
thank you so much :) i will try it today

Reply:
tested this version, working not good, skipping a lot of files.
webpack build around 200-300 files but triggered and printed to output only 10-12

Reply:
Can you please share a log?

Reply:
Can you please follow up?